Mango API/src/com/infiniteautomation/mango/rest/latest/FileStoreRestController.java

```diff
@@ -286,7 +289,11 @@ private FileModel fileToModel(FileStorePath fileStorePath, ServletContext contex
     }
 
     private Path findUniqueFileName(Path directory, String filename, boolean overwrite) {
-        Path file = directory.resolve(filename).toAbsolutePath().normalize();
+        // retrieve just the filename, removing path separators
+        Path pathFileName = Path.of(filename).getFileName();
+        filename = pathFileName.toString();
+
+        Path file = directory.resolve(pathFileName).toAbsolutePath().normalize();
```

Comment:
Am I reading this incorrectly, or does this remove the ability to write files in subdirectories of the filestore?

Reply:
No, the path to store the file in should be specified in the URL. It is passed in via https://github.com/MangoAutomation/ma-modules-public/blob/9ae7e73917520ab30aa0edc5b81ba3521d65f266/Mango%20API/src/com/infiniteautomation/mango/rest/latest/FileStoreRestController.java#L93C46-L93C57

Reply:
I did a functional test yesterday. 

I ran a new one just today:

Without the fix:
```
bperez@GDL-LAP-283:~/RAD-5473$ curl -v -k --location 'http://172.18.48.1:8085/rest/latest/file-stores/default?overwrite=true' --header 'Accept: application/json' --header "Authorization: Bearer $MANGO_TOKEN" --header 'Content-Type: multipart/form-data' --form 'file=@test.txt;filename=../test.txt'
*   Trying 172.18.48.1:8085...
* TCP_NODELAY set
* Connected to 172.18.48.1 (172.18.48.1) port 8085 (#0)
> POST /rest/latest/file-stores/default?overwrite=true HTTP/1.1
> Host: 172.18.48.1:8085
> User-Agent: curl/7.68.0
> Accept: application/json
> Authorization: Bearer eyJhbGciOiJFUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImV4cCI6MTcxNzY5NTE5MSwiaWQiOjEsInYiOjEsInR5cCI6ImF1dGgifQ.AXZBkem4C0SxhoEelTJm0Fn2lUo32SPwBJHLuRLmvQkYONA50-hhfdc79JLRo-DQi9HfVvLrJICCpmpbMBCV_oBYATb1DsDbau1j8NjVI3UDZ_ZpBInAxInkcDsPrLgznXIgA0ZACNpYGeE64IQesnt22-B3OmXzqiAoksl12n6-I79a
> Content-Length: 202
> Content-Type: multipart/form-data; boundary=------------------------91ceba4f0b775126
>
* We are completely uploaded and fine
* Mark bundle as not supporting multiuse
< HTTP/1.1 500 Server Error
< Date: Wed, 05 Jun 2024 17:33:41 GMT
< Cache-Control: no-store
< Content-Type: application/json;charset=utf-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< X-Frame-Options: SAMEORIGIN
< Transfer-Encoding: chunked
<
* Connection #0 to host 172.18.48.1 left intact
{"cause":"TranslatableIllegalArgumentException: Path is not a valid filestore path","mangoStatusCode":-1,"mangoStatusName":null,"localizedMessage":"Internal Server Error"}

```
**The file was uploaded to filestore**

**With the fix:**
```
bperez@GDL-LAP-283:~/RAD-5473$ curl -v -k --location 'http://172.18.48.1:8080/rest/latest/file-stores/default?overwrite=true' --header 'Accept: application/json' --header "Authorization: Bearer $MANGO_TOKEN" --header 'Content-Type: multipart/form-data' --form 'file=@test.txt;filename=../test.txt'
*   Trying 172.18.48.1:8080...
* TCP_NODELAY set
* Connected to 172.18.48.1 (172.18.48.1) port 8080 (#0)
> POST /rest/latest/file-stores/default?overwrite=true HTTP/1.1
> Host: 172.18.48.1:8080
> User-Agent: curl/7.68.0
> Accept: application/json
> Authorization: Bearer eyJhbGciOiJFUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImV4cCI6MTcxNzY5MjEyNiwiaWQiOjEsInYiOjEsInR5cCI6ImF1dGgifQ.AM75fiEFUMcX2hpisli6TkYmO4Jzz_l1yhTXKmUjY5yqbDC2mbE7Jjj10r1ToeMbtfPz3rpJQrO0yJoxekntYg4SAfxyRLwZtDpajU2b3MEvnI7rRwOkBpymzpHbHaXb7I1QWINC-hpZqOTn5titRioyq7jUd8YW6CipBcyw98gPvk1S
> Content-Length: 202
> Content-Type: multipart/form-data; boundary=------------------------6c3d6fc2d18679b8
>
* We are completely uploaded and fine
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Date: Wed, 05 Jun 2024 17:34:37 GMT
< Cache-Control: no-store
< Content-Type: application/json;charset=utf-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< X-Frame-Options: SAMEORIGIN
< Vary: Accept-Encoding
< Transfer-Encoding: chunked
<
[ {
  "folderPath" : "",
  "filename" : "test.txt",
  "relativePath" : "test.txt",
  "fileStoreXid" : "default",
  "mimeType" : "text/plain",
  "lastModified" : "2024-06-05T12:34:37.147-05:00",
  "size" : 13,
  "directory" : false
* Connection #0 to host 172.18.48.1 left intact
```

**The file was created in the directory filestore/default**